app/notes/[noteId]/page.tsx

```diff
@@ -9,15 +9,31 @@ type Props = {
 };
 export default async function NotePage({ params }: Props) {
   // Task: Restrict access to the note page only to the user who created the note
+
+  const noteId = Number((await params).noteId);
+
   // 1. Check if the sessionToken cookie exists
   const sessionTokenCookie = await getCookie('sessionToken');
 
-  // 2. Query the notes with the session token and noteId
+  // 2. Check if the note exists
+  const noteExists = await selectNoteExists(noteId);
+
+  if (!noteExists) {
```

Comment:
since this is a 1-time usage variable, we can inline it in the `if` condition

Reply:
bcca6eb391fe599825a4afa5bf1c87b835000b87